hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/CompactUpsertTargetDescriptor.java

```diff
@@ -26,29 +27,29 @@
 
 public class CompactUpsertTargetDescriptor implements UpsertTargetDescriptor {
 
-    private String typeName;
+    private Schema schema;
 
     @SuppressWarnings("unused")
     private CompactUpsertTargetDescriptor() {
     }
 
-    public CompactUpsertTargetDescriptor(@Nonnull String typeName) {
-        this.typeName = typeName;
+    public CompactUpsertTargetDescriptor(@Nonnull Schema schema) {
+        this.schema = schema;
     }
 
     @Override
     public UpsertTarget create(InternalSerializationService serializationService) {
-        return new CompactUpsertTarget(typeName);
+        return new CompactUpsertTarget(schema);
     }
 
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
-        out.writeString(typeName);
+        out.writeObject(schema);
```

Comment:
This may most probably be an a lot larger number of bytes serialized data. Where is this data transferred to? Would there be a negative cost of transferring a larger data?

Reply:
This is a must of the design. We want to create schema at the create mapping and distribute it to places where insert will happen